packages/core/src/middleware/koa-role-guard.ts

```diff
@@ -0,0 +1,22 @@
+import { UserRole } from '@logto/schemas';
+import { MiddlewareType } from 'koa';
+
+import RequestError from '@/errors/RequestError';
+import assertThat from '@/utils/assert-that';
+
+import { WithUserInfoContext } from './koa-user-info';
+
+export default function koaRoleGuard<
```

Comment:
can we put "role" as this functions first parameter? And it would be:
```ts
app.use(koaRoleGuard(UserRole.admin));
```

Reply:
updated